### DISCUSSION TITLE
How to set pages to a path that has a dynamic route ? eg : pages: {  signIn: '/[slug]/somepage', },

### DISCUSSION THREAD

**Discussion by rahul-269** [nextauthjs/next-auth#8899] [Help]
### Summary

I have a dynamic route before my actual pages , so it looks something like /[slug]/login or /[slug]/dashboard. 
I am trying to use the default middleware to protect my routes, and  I've set the middleware correctly. But in my next auth route file, I can't seem to figure out how to set my custom sign in page. I cannot set it to 

pages: {  signIn: '/[slug]/somepage', }

because it will display the slug as [slug] in the url. I cannot get the slug value here using useParams() either , so I cant put it into a variable and use that. It's my first time doing auth, and I can't seem to find a way to do this. I would appreciate any help in solving this , or maybe even alternative solutions. Thankyou

### Additional information

```js
default : pages: {
    signIn: '/login',
  },

//What I need 
pages: {
    signIn: '/slugvaluehere/login', 
  },
```

**Comment by janvorwerk** [comment]
Hello! 
I wish I could help, but I am struggling with the same kind of problem here... 
I am struggling like hell as soon as I am moving away from the default _/api/auth path_. Not sure what is the recommended way to have an internationalized signin page... 🥹

I am **not** using the middleware FWIW.

**Comment by janvorwerk** [comment]
I found out how to deal with internationalization of the login page. If this can help:

https://github.com/janvorwerk/next-auth-login-form

**Reply by rahul-269** [reply]
I'll have a look right now , thanks for the help !

**Reply by rahul-269** [reply]
Okay, so this doesn't help my use case per say, but this is valuable to anyone with a fixed number of possible dynamic route name values.

**Reply by janvorwerk** [reply]
Not sure I understand your use-case of the slug, and I don't understand your point of the fixed number of dynamic routes.
It's rather that the slug needs to be computable.
Actually, if the dynamic route was the date (`/2023-10-23/login` for instance), my solution would still work: The pages would be setup with `/login` and the middleware would be able to dynamically redirect to `/2023-10-23/login`).

**Comment by 418e** [comment]
I don't know if it will work but I guess you could try using regex: `/[A-Za-z]+/login`

**Reply by janvorwerk** [reply]
I would be surprised if this helped. My understanding of the `pages` setting, is that it tells the next-auth runtime where to redirect. So it should be a simple path.
In my [example](https://github.com/janvorwerk/next-auth-login-form/blob/d1ba5dcf12b5a25b6bdfa55889710c2e50e9ea6c/app/_lib/auth-options.ts#L6), although the signin page is under `/[lang]/login`, I configured the path to be just `/login`, and it's a middleware that does the job of redirecting `/login` to `/en/login` (when the selected/detected language is English).
Hope this helps!

**Reply by rahul-269** [reply]
Correct , it should be a simple path. /[A-Za-z]+/login will appear in the url as is if you try to do it and it will look something like yourdomain/[A-Za-z]+/login in your browser.
